File: AssignmentSystemApp/src/Services/authService.js
import { loginApi } from "../api/authApi";

export const loginService = async ({ email, password }) => {
    try {
        if (!email?.trim()) { throw new Error("Email is required."); }
        if (!password?.trim()) { throw new Error("Password is required."); }
        const response = await loginApi({email, password,});
        if (!response.success) {
            throw new Error(response.message || "Login failed.");
        }
        return {
            token: response.data.token,
            expiration: response.data.expiration,
            user: {
                id: response.data.userId,
                firstName: response.data.firstName,
                lastName: response.data.lastName,
                email: response.data.email,
                role: response.data.role,
            },
            message: response.message,
        };
    } 
    catch (error) {
        throw new Error(
            error.response?.data?.message ||
            error.message ||
            "Unable to login."
        );
    }
};

File: AssignmentSystem/AssignmentSystem/Services/Implementations/AuthService.cs
using AssignmentSystem.DTOs.Auth;
using AssignmentSystem.Helpers;
using AssignmentSystem.Repositories.Interfaces;
using AssignmentSystem.Services.Interfaces;

namespace AssignmentSystem.Services.Implementations
{
    public class AuthService : IAuthService
    {
        private readonly IUnitOfWork _unitOfWork;
        private readonly IJwtService _jwtService;

        public AuthService(
            IUnitOfWork unitOfWork,
            IJwtService jwtService)
        {
            _unitOfWork = unitOfWork;
            _jwtService = jwtService;
        }

        public async Task<ApiResponse<LoginResponseDto>> LoginAsync(LoginRequestDto request)
        {
            var user = await _unitOfWork.Users.LoginAsync(request.Email, request.Password);

            if (user == null)
            {
                return new ApiResponse<LoginResponseDto>
                {
                    Success = false,
                    StatusCode = 401,
                    Message = "Invalid email or password."
                };
            }

            var token = _jwtService.GenerateToken(user);

            return new ApiResponse<LoginResponseDto>
            {
                Success = true,
                StatusCode = 200,
                Message = "Login successful.",
                Data = new LoginResponseDto
                {
                    Token = token,
                    Expiration = DateTime.UtcNow.AddMinutes(120),
                    UserId = user.Id,
                    FirstName = user.FirstName,
                    LastName = user.LastName,
                    Email = user.Email,
                    Role = user.Role.Name
                }
            };
        }
    }
}
